logical/plugin/backend_client.go

```diff
@@ -0,0 +1,226 @@
+}
+
+// TypeReply is the reply for the Type method.
+type TypeReply struct {
+	Type logical.BackendType
+}
+
+func (b *backendPluginClient) HandleRequest(req *logical.Request) (*logical.Response, error) {
+	// Shim logical.Storage
+	id := b.broker.NextId()
+	go b.broker.AcceptAndServe(id, &StorageServer{
+		impl: req.Storage,
+	})
+
+	argReq := new(logical.Request)
+	*argReq = *req
+	argReq.Storage = nil
+
+	args := &HandleRequestArgs{
+		StorageID: id,
+		Request:   argReq,
+	}
+	var reply HandleRequestReply
+
+	err := b.client.Call("Plugin.HandleRequest", args, &reply)
+	if err != nil {
+		return nil, err
+	}
+	if reply.Error != nil {
+		if reply.Error.Error() == logical.ErrUnsupportedOperation.Error() {
+			return nil, logical.ErrUnsupportedOperation
+		}
+		return nil, reply.Error
+	}
+
+	return reply.Response, nil
+}
+
+func (b *backendPluginClient) SpecialPaths() *logical.Paths {
+	var reply SpecialPathsReply
+	// var paths logical.Paths
+	err := b.client.Call("Plugin.SpecialPaths", new(interface{}), &reply)
+	if err != nil {
+		return nil
+	}
+
+	return reply.Paths
+}
+
+// System returns vault's system view. The backend client stores the view during
+// Configure, so there is no need to shim the system just to get it back.
+func (b *backendPluginClient) System() logical.SystemView {
+	return b.system
+}
+
+// Logger returns vault's logger. The backend client stores the logger during
+// Configure, so there is no need to shim the logger just to get it back.
+func (b *backendPluginClient) Logger() log.Logger {
+	return b.logger
+}
+
+func (b *backendPluginClient) HandleExistenceCheck(req *logical.Request) (bool, bool, error) {
+	// Shim logical.Storage
+	id := b.broker.NextId()
+	go b.broker.AcceptAndServe(id, &StorageServer{
+		impl: req.Storage,
+	})
+
+	argReq := new(logical.Request)
+	*argReq = *req
+	argReq.Storage = nil
+
+	args := &HandleExistenceCheckArgs{
+		StorageID: id,
+		Request:   argReq,
+	}
+	var reply HandleExistenceCheckReply
+
+	err := b.client.Call("Plugin.HandleExistenceCheck", args, &reply)
+	if err != nil {
+		return false, false, err
+	}
+	if reply.Error != nil {
+		// THINKING: Should be be a switch on all error types?
+		if reply.Error.Error() == logical.ErrUnsupportedPath.Error() {
+			return false, false, logical.ErrUnsupportedPath
+		}
+		return false, false, reply.Error
+	}
+
+	return reply.CheckFound, reply.Exists, nil
+}
+
+func (b *backendPluginClient) Cleanup() {
+	b.client.Call("Plugin.Cleanup", new(interface{}), &struct{}{})
+}
+
+func (b *backendPluginClient) Initialize() error {
+	err := b.client.Call("Plugin.Initialize", new(interface{}), &struct{}{})
+	return err
+}
+
+func (b *backendPluginClient) InvalidateKey(key string) {
+	b.client.Call("Plugin.InvalidateKey", key, &struct{}{})
+}
+
+func (b *backendPluginClient) Configure(config *logical.BackendConfig) error {
+	// Shim logical.Storage
+	storageID := b.broker.NextId()
+	go b.broker.AcceptAndServe(storageID, &StorageServer{
```

Comment:
I think the comment you're referring to wasn't me saying there are other places where cleanup is done, but asking if there should be (the missing defer'd close calls in handlerequest/handleexistencecheck). So I think it's fine.